__tests__/find-ruby.test.ts

```diff
@@ -1,74 +1,67 @@
-import * as io from '@actions/io';
+import * as tc from '@actions/tool-cache';
+import * as core from '@actions/core';
 import fs = require('fs');
 import os = require('os');
 import path = require('path');
+import * as cache from '../src/cache';
+import {run} from '../src/main';
 
-const toolDir = path.join(__dirname, 'runner', 'tools');
-const tempDir = path.join(__dirname, 'runner', 'temp');
+describe('find-ruby', () => {
+  let inSpy: jest.SpyInstance;
+  let tcSpy: jest.SpyInstance;
+  let cnSpy: jest.SpyInstance;
 
-process.env['AGENT_TOOLSDIRECTORY'] = toolDir;
-process.env['RUNNER_TOOL_CACHE'] = toolDir;
-process.env['RUNNER_TEMP'] = tempDir;
+  //beforeAll(async () => {});
 
-import {findRubyVersion} from '../src/installer';
+  beforeEach(() => {
+    tcSpy = jest.spyOn(tc, 'find');
+    inSpy = jest.spyOn(core, 'getInput');
+    cnSpy = jest.spyOn(process.stdout, 'write');
+    cnSpy.mockImplementation(line => {
+      // uncomment to debug
+      // process.stderr.write('write2:' + line + '\n');
+    });
+  });
 
-describe('find-ruby', () => {
-  beforeAll(async () => {
-    await io.rmRF(toolDir);
-    await io.rmRF(tempDir);
+  afterEach(() => {
+    tcSpy.mockClear();
+    cnSpy.mockClear();
+    jest.clearAllMocks();
   });
 
-  afterAll(async () => {
-    try {
-      await io.rmRF(toolDir);
-      await io.rmRF(tempDir);
-    } catch {
-      console.log('Failed to remove test directories');
-    }
-  }, 100000);
+  //afterAll(async () => {}, 100000);
 
-  it('Uses version of ruby installed in cache', async () => {
-    const rubyDir: string = path.join(toolDir, 'Ruby', '17.0.0', os.arch());
-    await io.mkdirP(rubyDir);
-    fs.writeFileSync(`${rubyDir}.complete`, 'hello');
-    // This will throw if it doesn't find it in the cache (because no such version exists)
-    await findRubyVersion('17.0.0');
-  });
+  it('finds a version of ruby already in the cache', async () => {
+    tcSpy.mockImplementation(() => '/cache/ruby/2.7.0');
 
-  it('Uses 9.0.x version of ruby installed in cache ', async () => {
-    const rubyDir: string = path.join(toolDir, 'Ruby', '9.0.1', os.arch());
-    await io.mkdirP(rubyDir);
-    fs.writeFileSync(`${rubyDir}.complete`, 'hello');
-    // This will throw if it doesn't find it in the cache (because no such version exists)
-    await findRubyVersion('9.0.x');
+    let cacheDir: string = await cache.find('2.7.0');
+    expect(cacheDir).toBe(normalize('/cache/ruby/2.7.0/bin'));
```

Comment:
had to normalize what to expect because of windows.  cache.find does path.combine which will flip windows delims

Reply:
there is a `path.normalize()`, could we use that?

Reply:
path.normalize is about resolving a series of paths (effectively a series of CD statements) which doesn't do what this does.

Reply:
```
H:\Source\c2c-actions-runtime\adrs>node
Welcome to Node.js v12.10.0.
Type ".help" for more information.
> path.normalize('/foo/bar/baz')
'\\foo\\bar\\baz'
> 
```

Reply:
we're also fixing of line endings.  perhaps the normalize function should call path.normalize.  I will try

Reply:
it would be a little bit weird the dir has line ending. :D

Reply:
see the last test.  It's ::add-path::xxxx which has a line ending.  We're expecting an output line.  I could split into two things ...

Reply:
I renamed the function to fixPaths and it's also fixing up strings which have command output strings.